src/hardware/vga_draw.cpp

```diff
@@ -777,9 +717,10 @@ static void VGA_DrawSingleLine(uint32_t /*blah*/)
 			// so use black for this case too...
 			//if (vga.attr.disabled& 2) {
 			if (vga.dac.xlat16[bg_color_index] != 0) {
-				for (Bitu i = 0; i < 256; ++i)
+				for (uint16_t i = 0; i < 256; ++i)
```

Comment:
can this be a `uint8_t` to avoid the cast below, and there's the assert as well?

Reply:
Yeah - these interesting - the counter indexes into an array with a max-element of [255], however when `i` hits 255, the `++i` wraps it back to `0` and the condition never triggers.

Hmm - maybe `uint8_t` is still doable with a while-loop? 

```c++
uint8_t i = UINT8_MAX;
while (i != 255) {
  ++i;
  //do stuff with i
}
```

(Maybe I'm making this harder that it is?)

Reply:
The hardcoded '256' limit comes from having to loop over the xlat16 table. But if the translation table changes, then this code will silently break, especially if the table is made bigger.

Now we directly get the array length and also check the assumption about being 256 long.
```
	// check some assumptions about the translation table
	constexpr auto xlat16_len = ARRAY_LENGTH(vga.dac.xlat16);
	static_assert(xlat16_len == 256);

	for (auto i = 0; i < xlat16_len; ++i)
		...
```